apps/css-workshop/src/components/Icon.astro

```diff
@@ -0,0 +1,46 @@
+---
+type Props = {
+  svg: string;
+  size?: 's' | 'm' | 'l' | 'auto';
+  wrapperProps?: astroHTML.JSX.HTMLAttributes;
+  buttonIcon?: boolean;
+} & astroHTML.JSX.HTMLAttributes;
+
+const {
+  svg,
+  size = 'm',
+  class: className,
+  buttonIcon = false,
+  wrapperProps,
+  ...rest
+} = Astro.props;
+
+const iconProps = {
+  ['class:list']: [
+    {
+      ['iui-svg-icon']: !buttonIcon,
+    },
+    className,
+  ],
+  ['data-iui-icon-size']: size,
+  ['aria-hidden']: 'true',
+  ...rest,
+};
+
+const basic = !Astro.slots.has('default');
+const Element = `svg-${svg}`;
```

Comment:
Instead of the `basic` prop, I think it would be better to create two different components: `Icon` and `Svg`, because they do different things.
- `Icon`: Only apply `iui-svg-icon` class and `data-iui-icon-size` attributes.
- `Svg`: Only load the correct svg from `@itwin/itwinui-icons-elements`.

Reply:
Actually I think I might be misunderstanding the `basic` prop. Why exactly is it necessary?